app/src/main/java/com/techcourse/dao/UserDao.java

```diff
@@ -1,121 +1,81 @@
 package com.techcourse.dao;
 
-import com.techcourse.domain.User;
 import com.interface21.jdbc.core.JdbcTemplate;
+import com.interface21.jdbc.core.RowMapper;
+import com.techcourse.domain.User;
+import java.util.List;
+import javax.sql.DataSource;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import javax.sql.DataSource;
-import java.sql.Connection;
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.util.List;
-
 public class UserDao {
 
     private static final Logger log = LoggerFactory.getLogger(UserDao.class);
 
     private final DataSource dataSource;
+    private final JdbcTemplate jdbcTemplate;
 
     public UserDao(final DataSource dataSource) {
         this.dataSource = dataSource;
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
     }
 
     public UserDao(final JdbcTemplate jdbcTemplate) {
         this.dataSource = null;
+        this.jdbcTemplate = jdbcTemplate;
     }
 
+    private static final RowMapper<User> USER_ROW_MAPPER = (resultSet) -> new User(
```

Comment:
객체 변환 책임을 별도로 분리한 부분 👍 👍 

그런데  수양은 이 `USER_ROW_MAPPER`가 상수라고 생각하시나요?  아니면 정적 객체라고 생각하시나요? 
저는 정적 객체라는 생각이 들어서 `log`와 동일한 컨벤션으로 가져가는 게 자연스러울 것 같아요!
(선언 순서, 스네이크 케이스 -> 카멜 케이스)

```suggestion
    private static final Logger log = LoggerFactory.getLogger(UserDao.class);
    private static final RowMapper<User> userRowMapper = (resultSet) -> new User(
```